userspace/libsinsp/sinsp.cpp

```diff
@@ -697,6 +729,15 @@ void sinsp::open_test_input(scap_test_input_data* data)
 
 /*=============================== OPEN METHODS ===============================*/
 
+/*=============================== Engine related ===============================*/
+
+bool sinsp::check_current_engine(const std::string& engine_name)
+{
+	return m_engine_name.compare(engine_name) == 0;
```

Comment:
I don't get why we would want to store `m_engine_name` in sinsp; isn't it better if we had like a `scap_get_engine_name` api that calls a vtable method on the current engine?

Reply:
yeah, we can do that!